Source/Csla.Blazor/State/StateManager.cs

```diff
@@ -0,0 +1,191 @@
+    /// <param name="firstRender">firstRender value from OnAfterRender method</param>
+    /// <returns></returns>
+    /// <remarks>
+    /// This is called in OnAfterRender methods of a Blazor page (sync
+    /// or async). This way the page can be flexibly
+    /// rendered in server-rendered, server-interactive, and wasm-interactive
+    /// modes and the state will always be available.
+    /// </remarks>
+    public async Task AfterRender(bool firstRender)
+    {
+      await AfterRender(firstRender, TimeSpan.FromSeconds(10));
+    }
+
+    /// <summary>
+    /// Get state from web server for use in Blazor wasm.
+    /// client.
+    /// </summary>
+    /// <param name="firstRender">firstRender value from OnAfterRender method</param>
+    /// <param name="timeout">Time to wait before timing out</param>
+    /// <returns></returns>
+    /// <remarks>
+    /// This is called in OnAfterRender methods of a Blazor page (sync
+    /// or async). This way the page can be flexibly
+    /// rendered in server-rendered, server-interactive, and wasm-interactive
+    /// modes and the state will always be available.
+    /// </remarks>
+    public async Task AfterRender(bool firstRender, TimeSpan timeout)
+    {
+      await GetState(CalledFrom.AfterRender, firstRender, timeout);
+    }
+
+    /// <summary>
+    /// Get state from web server for use in Blazor wasm. Must call
+    /// as user navigates from any server-side Blazor page to a wasm
+    /// Blazor page.
+    /// client.
+    /// </summary>
+    /// <param name="calledFrom">Location method is invoked in page</param>
+    /// <param name="firstRender">firstRender value from OnAfterRender method</param>
+    /// <param name="timeout">Time to wait before timing out</param>
+    /// <returns></returns>
+    /// <remarks>
+    /// Normally this is called in both the initialize and after render
+    /// methods of a Blazor page. This way the page can be flexibly
+    /// rendered in server-rendered, server-interactive, and wasm-interactive
+    /// modes and the state will always be available.
+    /// </remarks>
+    private async Task GetState(CalledFrom calledFrom, bool firstRender, TimeSpan timeout)
+    {
+      var isBrowser = OperatingSystem.IsBrowser();
+      if (isBrowser)
+      {
+        if (calledFrom == CalledFrom.Initialize && firstRender)
+        {
+          var session = await _sessionManager.RetrieveSession();
+          IsStateAvailable = session.IsFullyInitialized;
+        }
+      }
+      else
+      {
+        if (calledFrom == CalledFrom.Initialize)
+        {
+          await WaitForState(timeout);
+        }
+      }
+    }
+
+    /// <summary>
+    /// Waits for state to be available as it is transferred
+    /// from Blazor wasm to the web server.
+    /// </summary>
+    /// <param name="timeout">Time to wait before timing out</param>
+    private async Task WaitForState(TimeSpan timeout)
+    {
+      var session = _sessionManager.GetSession();
```

Comment:
Could this be a problem? Here gets a _new_ session created and not the already existing one reused.